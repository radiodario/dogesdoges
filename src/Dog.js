import React, { Component } from 'react';

class Dog extends Component {
  constructor(props) {
    super(props)
    this.state = {
      image : null,
      text : '',
    }
  }

  componentWillMount() {
    this.loadText();
    this.loadImage();
  }

  async loadImage() {
    const { breed, subBreed } = this.props;

    let breedName = breed
    if (subBreed) {
      breedName += `/${subBreed}`
    }

    const imageDataUrl = `https://dog.ceo/api/breed/${breedName}/images/random`

    try {
      const response = await fetch(imageDataUrl)
      const { message } = await response.json()
      this.setState({ image : message})
    } catch (e) {
      return e.message;
    }
  }

  async loadText() {
    const url = 'https://baconipsum.com/api/?type=all-meat&sentences=1&start-with-lorem=1'

    try {
      const response = await fetch(url)
      const textStrings = await response.json()
      this.setState({ text : textStrings.join(' ')})
    } catch (e) {
      return e.message;
    }
  }


  render() {
    const { breed, subBreed } = this.props;
    const { image, text } = this.state;

    const breedName = subBreed ? `${breed} - ${subBreed}` : breed

    return (
      <div className="Dog-Breed">
        <h4>{ breedName }</h4>
        <img src={image} alt={ breedName }/>
        <div className="Dog-Breed--Text">
          { text }
        </div>
      </div>
    )

  }

}


export default Dog;
