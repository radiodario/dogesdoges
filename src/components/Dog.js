import React from 'react';
import PropTypes from 'prop-types';

const Dog = ({ breed, subBreed, image, text }) => {
  const breedName =  subBreed ? `${breed} - ${subBreed}` : breed
  return (
    <div className="Dog-Breed">
      <h4>{ breedName }</h4>
      <img
        src={image}
        alt={breedName}
      />
      <div className="Dog-Breed--Text">
        { text }
      </div>
    </div>
  )
}

Dog.propTypes = {
  breed: PropTypes.string.isRequired,
  subBreed: PropTypes.string,
  image: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
}

export default Dog;
