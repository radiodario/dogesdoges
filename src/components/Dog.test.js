import React from 'react';
import Dog from './Dog';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer
    .create(<Dog
      breed="corgi"
      subBreed="pembroke"
      image="https://placehold.it/100"
      text="woof woof"/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

