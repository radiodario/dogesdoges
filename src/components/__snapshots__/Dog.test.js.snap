// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="Dog-Breed"
>
  <h4>
    corgi - pembroke
  </h4>
  <img
    alt="corgi - pembroke"
    src="https://placehold.it/100"
  />
  <div
    className="Dog-Breed--Text"
  >
    woof woof
  </div>
</div>
`;
